web/src/lib/components/forms/library-user-picker-form.svelte

```diff
@@ -37,6 +39,7 @@
     <p class="p-5 text-sm">NOTE: This cannot be changed later!</p>
 
     <SettingSelect bind:value={ownerId} options={userOptions} name="user" />
+    <SettingSwitch title="Read Only" subtitle="Disable deletion and modification" bind:checked={isReadOnly} />
```

Comment:
does this disable modification? I saw you updated `handleAssetDeletion` in `asset.service.ts`, but I did not see changes to any other methods such as `update`

Reply:
There are no server-side checks of isExternal or isReadOnly before modification. I have modified checks on web-side. It was tested, modification works, creates sidecar file in external lib.

Reply:
Should the text here be updated to just say "Disable deletion" instead of "Disable deletion and modification" then?

Reply:
It disables modifications because frontend checks isReadOnly.

Reply:
Ah, I misunderstood. It seems like it should also disable it on the server. Otherwise you can modify via the mobile app or third-party integrations using the API. It'd be weird to only disallow it in the web frontend, but allow it from these other locations.

Reply:
Disabled deletion and modification for External Library's assets is default behavior of immich.

This PR allows this actions by disabling ReadOnly. Now I can disable highlighted switch and cleanup my External Library. When I started this fork, I looked at asset.service and didn't find any restrictions, then tested and it worked. 

I agree that frontend-only restrictions is not okay, but think that it's not this PR's problem. I can enforce server-side checks but in another PR.

You can't _see changes to any other methods such as update_ because it is not required for this task. I made minimal changes allowing deletion and modification of ExtLib assets. 

Regards